frontend/packages/shared/src/queryInvalidator/SyncSuccessQueriesInvalidator.test.ts

```diff
@@ -0,0 +1,45 @@
+import { QueryClient } from '@tanstack/react-query';
+import { SyncSuccessQueriesInvalidator } from './SyncSuccessQueriesInvalidator';
+import { QueryKey } from 'app-shared/types/QueryKey';
+import { waitFor } from '@testing-library/react';
+
+jest.mock('@tanstack/react-query');
+
+describe('SyncSuccessQueriesInvalidator', () => {
+  let queryClientMock: QueryClient;
+  const org = 'testOrg';
+  const app = 'testApp';
+
+  beforeEach(() => {
+    queryClientMock = new QueryClient();
+    queryClientMock.invalidateQueries = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should invalidate query cache only once when invalidateQueryByFileName is called', async () => {
+    const instance = SyncSuccessQueriesInvalidator.getInstance(queryClientMock, org, app);
+
+    const fileName = 'applicationmetadata.json';
+    instance.invalidateQueryByFileName(fileName);
```

Comment:
Should we in this test try to add the same filename twice and then verify the invalidation happens only once?
```suggestion
    const fileName = 'applicationmetadata.json';
    instance.invalidateQueryByFileName(fileName);
    instance.invalidateQueryByFileName(fileName);
```